### PR TITLE
Attribute sort specification

### PR DESCRIPTION
**Pull request by jake-bassett**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

**Review by Exac — APPROVED** [review]
Quick question: I noticed that in some of our code we have three sort directions, `'ASC'`, `'DESC'`, and `undefined`. Since we are asserting that the `direction` is always present, I suppose we don't have to worry about that in this particular model?

Otherwise LGTM.

File: projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  @ModelProperty({
+    key: 'key',
+    type: STRING_PROPERTY.type,
+    required: true
+  })
+  public key!: string;
+
+  @ModelProperty({
+    key: 'direction',
+    type: STRING_PROPERTY.type,
```
Comment (itssharmasandeep) [inline projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts:22]: we need to mark it as required if it is

File: projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { GraphQlSortDirection } from '../../../../graphql/model/schema/sort/graphql-sort-direction';
+import { Specification } from '../../../../graphql/model/schema/specifier/specification';
+import { SpecificationBuilder } from '../../../../graphql/request/builders/specification/specification-builder';
+import { SpecificationModel } from './specification.model';
+
+@Model({
+  type: 'attribute-sort-specification',
+  displayName: 'Attribute Sort'
+})
+export class AttributeSortSpecificationModel extends SpecificationModel<Specification> {
```
Comment (itssharmasandeep) [inline projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts:12]: Confused about why we need this to extend the specification model?

File: projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts (hunk 1 above)
Comment (palbizu) [reply projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts:22]: direction could be undefined IMO (when is not applied). Do you think that should be mandatory anyways?

File: projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts (hunk 2 above)
Comment (jake-bassett) [reply projects/observability/src/shared/dashboard/data/graphql/specifiers/attribute-sort-specification.model.ts:12]: You are the man. Thanks for calling this out. Its unnecessary for this model.

**Comment by jake-bassett** [comment]
> Quick question: I noticed that in some of our code we have three sort directions, `'ASC'`, `'DESC'`, and `undefined`. Since we are asserting that the `direction` is always present, I suppose we don't have to worry about that in this particular model?
> 
> Otherwise LGTM.

Yeah, in this case just the omission of the model itself is the third state. If no sorting was required, this model would never be invoked.

**Pull request closed without merging**